python/cudf/cudf/core/multiindex.py

```diff
@@ -870,18 +824,17 @@ def _index_and_downcast(self, result, index, index_key):
             # Pandas returns an empty Series with a tuple as name
             # the one expected result column
             series_name = []
-            for code in index._source_data.columns:
-                series_name.append(index._source_data[code][0])
+            for code in index.names:
+                series_name.append(index._data[code][0])
```

Comment:
There is a possibility that `names` of `MultiIndex` can contain duplicates, in that case, this would silently append the same column no?